src/platform/platform_posix.c

```diff
@@ -658,6 +658,12 @@ CxPlatThreadCreate(
             "[ lib] ERROR, %u, %s.",
             Status,
             "pthread_create failed");
+
+        // Try it again without attributes.
+        // pthread_create can fail with ENOKEY if we request affinity on CPU that is offline
+        if (Status == ENOKEY && pthread_create(Thread, NULL, Config->Callback, Config->Context) == 0) {
+            Status = QUIC_STATUS_SUCCESS;
```

Comment:
We should still be checking `errno` just in case it is failing for some other reason.

Reply:
I can return new errno in `Status` or I can spit out another trace message. Any preference?

Reply:
How about change the code to the following?
```c
    //
    // If pthread_create fails eith ENOKEY, then try again without the attribute
    // because the CPU might be offline.
    //
    if (pthread_create(Thread, &Attr, CxPlatThreadCustomStart, CustomContext) &&
        errno != ENOKEY || pthread_create(Thread, NULL, CxPlatThreadCustomStart, CustomContext)) {
        Status = errno;
        QuicTraceEvent(
            LibraryErrorStatus,
            "[ lib] ERROR, %u, %s.",
            Status,
            "pthread_create failed");
        CXPLAT_FREE(CustomContext, QUIC_POOL_CUSTOM_THREAD);
    }
```

Reply:
(well, I copied the wrong version, but you get the idea)

Reply:
```c
    //
    // If pthread_create fails eith ENOKEY, then try again without the attribute
    // because the CPU might be offline.
    //
    if (pthread_create(Thread, &Attr, Config->Callback, Config->Context) &&
        (errno != ENOKEY || pthread_create(Thread, NULL, Config->Callback, Config->Context))) {
        Status = errno;
        QuicTraceEvent(
            LibraryErrorStatus,
            "[ lib] ERROR, %u, %s.",
            Status,
            "pthread_create failed");
    }
```